google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionClient.java

```diff
@@ -198,7 +198,7 @@ DatabaseId getDatabaseId() {
   }
 
   /** Create a single session. */
-  SessionImpl createSession() {
+  SessionInstance createSession() {
```

Comment:
@olavloite Presently I have not modified this for multiplexed session method. If you like the approach, I can do a bit more refactoring to ensure that the session creation flow (control plane) only uses the new `SessionInstance` class and we use `SessionImpl` on the data-plane side of things. Also I haven't modified the class docs as of now. Will do after we finalize the approach, naming conventions, etc.